### PR TITLE
Make validation placeholders always available

### PR DESCRIPTION
**Pull request by jreklund**
This are another take on #2817 and the PR #2819 to fix it.

There have been a couple of posts in the forum wanting/trying to use placeholders, but it's limited to Models. It's such a nice feature that it should be available in any form of validation.

Now people can e.g. store a rule in their config file and pass it along from the controller.

- Moved fillPlaceholders from Model into Validation
- Added test for is_unique with placeholder. I think one are enough, it either works or it dosen't with `$this->validation->run()`, as all function depending on validation are ending up in the the run() function.
- Moved "Validation Placeholders" from the Model userguide into Validation.
- "Translation Of Messages And Validation Labels" never got rendered in Validation userguide.
- Fixed some indention in Validation userguide. "tabs" gives 8 spaces, instead of 4.

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [X] Unit testing, with >80% coverage
- [X] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: system/Model.php [hunk 1]
```diff
@@ -1473,62 +1469,6 @@ protected function cleanValidationRules(array $rules, array $data = null): array
...
-	 * The value of {id} would be replaced with the actual id in the form data:
-	 *
-	 *  'required|is_unique[users,email,id,13]'
-	 *
-	 * @param array $rules
-	 * @param array $data
-	 *
-	 * @return array
-	 */
-	protected function fillPlaceholders(array $rules, array $data): array
```
Comment (samsonasik) [inline system/Model.php:1496]: I think direct remove protected method is BC break. To keep it, it need to be kept with with change like the following:

```php
    /*
     * @deprecated use fillPlaceholders($data) from Validation instead
     */
    protected function fillPlaceholders(array $rules, array $data): array
    {
        return (function ($validation, $data) {
            $validation->fillPlaceholders($data);
            return $validation->getRules();
        })->bindTo($validation = $this->validation, get_class($this->validation))($validation, $data);
    }
```

Comment (jreklund) [reply system/Model.php:1496]: No other functions in CodeIgniter call fillPlaceholders. You wan't this in case someone extended Model?

Comment (samsonasik) [reply system/Model.php:1496]: That's the cost of `protected` method instead of `private`. Ok, let's the maintainer decide to keep or remove it ;)

Comment (lonnieezell) [reply system/Model.php:1496]: Yeah, I have to agree with @samsonasik here. Anyone extending could use it directly. Even though not too many probably are, we should leave it and add the deprecated note, I think.

**Comment by lonnieezell** [comment]
@jreklund After the deprecation notice mentioned above, I think we're good to include this one. Nice change. Thanks.

**Comment by jreklund** [comment]
Hi, thanks for the feedback. I will have the changes ready at the end of the week ready for CI 4.0.3 release.

**Comment by samsonasik** [comment]
@jreklund also, please rebas against latest develop.

**Comment by jreklund** [comment]
Damn... that looks so weird doing a rebase... Dosen't look right at all!

**Pull request closed without merging**